File: Week_6/problem_6-Matrix.py
# https://codeforces.com/group/MWSDmqGsZm/contest/219774/problem/T

n = int(input())
ar = [] # Initialize an empty list to store the matrix

for i in range(n):
    # Read a row of the matrix 
    # Eaxmple: 1 2 3 for first i
    # then 4, 5, 6 for first i
    # &, finaly 7, 8, 9 for last i
    row = list(map(int, input().split()))

    # Append the row to the matrix list
    ar.append(row)


primary_diagonal_sum = 0
secondary_diagonal_sum = 0


# assume i for row and j foe column traversal
for i in range(n):
    for j in range(n):
        if i == j:
            primary_diagonal_sum += ar[i][j]
        if i + j == n - 1:
            secondary_diagonal_sum += ar[i][j]


print(abs(primary_diagonal_sum - secondary_diagonal_sum))

File: Week_6/problem_6-Matrix.c
// https://codeforces.com/group/MWSDmqGsZm/contest/219774/problem/T

#include <stdio.h>
#include <limits.h>
#include <string.h>
#include <math.h>
#include <stdlib.h>

int main()
{
    int n;
    scanf("%d", &n);

    int row = n, col = n;
    int ar[row][col];

    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j < col; j++)
        {
            scanf("%d", &ar[i][j]);
        }
    }

    int primary_diagonal_sum = 0, secondary_diagonal_sum = 0;

    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j < col; j++)
        {
            if (i == j)
            {
                primary_diagonal_sum += ar[i][j];
            }
            if (i + j == n - 1)
            {
                secondary_diagonal_sum += ar[i][j];
            }
        }
    }

    // printf("%d %d", primary_diagonal_sum, secondary_diagonal_sum);

    printf("%d", abs(primary_diagonal_sum - secondary_diagonal_sum));

    return 0;
}
